tests/test_edge_cases.py

```diff
@@ -1506,3 +1524,108 @@ class Model(BaseModel):
         'nullable1': 1,
         'nullable2': 'two',
     }
+
+
+def test_custom_generic_validators():
+    T1 = TypeVar('T1')
+    T2 = TypeVar('T2')
+
+    class MyGen(Generic[T1, T2]):
+        def __init__(self, t1: T1, t2: T2):
+            self.t1 = t1
+            self.t2 = t2
+
+        @classmethod
+        def __get_validators__(cls):
+            yield cls.validate
+
+        @classmethod
+        def validate(cls, v, field):
+            if not isinstance(v, cls):
+                raise ValueError('Invalid value')
+            errors = []
+            for f, name in zip(field.sub_fields, ['t1', 't2']):
```

Comment:
this seems more involved than the approach used in the docs, could we simplify?

Reply:
Sure thing.